crates/api_models/src/admin.rs

```diff
@@ -2003,13 +2015,25 @@ pub struct BusinessProfileCreate {
     /// Whether to use the billing details passed when creating the intent as payment method billing
     pub use_billing_as_payment_method_billing: Option<bool>,
 
-    /// A boolean value to indicate if customer shipping details needs to be collected from wallet connector (Eg. Apple Pay, Google Pay, etc.)
+    /// A boolean value to indicate if customer shipping details needs to be collected from wallet
+    /// connector only if it is required field for connector (Eg. Apple Pay, Google Pay etc)
+    #[schema(default = false, example = false)]
+    pub collect_shipping_details_from_wallet_connector_if_required: Option<bool>,
+
+    /// A boolean value to indicate if customer billing details needs to be collected from wallet
+    /// connector only if it is required field for connector (Eg. Apple Pay, Google Pay etc)
     #[schema(default = false, example = false)]
-    pub collect_shipping_details_from_wallet_connector: Option<bool>,
+    pub collect_billing_details_from_wallet_connector_if_required: Option<bool>,
 
-    /// A boolean value to indicate if customer billing details needs to be collected from wallet connector (Eg. Apple Pay, Google Pay, etc.)
+    /// A boolean value to indicate if customer shipping details needs to be collected from wallet
+    /// connector irrespective of connector required fields (Eg. Apple pay, Google pay etc)
     #[schema(default = false, example = false)]
-    pub collect_billing_details_from_wallet_connector: Option<bool>,
+    pub always_collect_shipping_details_from_wallet_connector: Option<bool>,
```

Comment:
On a conceptual level, won't this field overwrite the behavior of `collect_shipping_details_from_wallet_connector_if_required`?

Then shouldn't there be some check to verify the same?